zio-http/src/main/scala/zio/http/netty/EventLoopGroups.scala

```diff
@@ -0,0 +1,67 @@
+package zio.http
+package netty
+
+import io.netty.channel._
+import io.netty.channel.epoll.{Epoll, EpollEventLoopGroup}
+import io.netty.channel.kqueue.{KQueue, KQueueEventLoopGroup}
+import io.netty.channel.nio.NioEventLoopGroup
+import io.netty.incubator.channel.uring.IOUringEventLoopGroup
+import zio._
+
+import java.util.concurrent.Executor
+
+/**
+ * Simple wrapper over NioEventLoopGroup
+ */
+object EventLoopGroups {
+
+  trait Config extends ChannelType.Config {
+    def nThreads: Int
+  }
+
+  def nio(nThreads: Int): ZIO[Scope, Nothing, EventLoopGroup] =
+    make(ZIO.succeed(new NioEventLoopGroup(nThreads)))
+
+  def nio(nThreads: Int, executor: Executor): ZIO[Scope, Nothing, EventLoopGroup] =
+    make(ZIO.succeed(new NioEventLoopGroup(nThreads, executor)))
+
+  def make(eventLoopGroup: UIO[EventLoopGroup]): ZIO[Scope, Nothing, EventLoopGroup] =
+    ZIO.acquireRelease(eventLoopGroup)(ev => NettyFutureExecutor.unit(ev.shutdownGracefully).orDie)
+
+  def epoll(nThreads: Int): ZIO[Scope, Nothing, EventLoopGroup] =
+    make(ZIO.succeed(new EpollEventLoopGroup(nThreads)))
+
+  def kqueue(nThreads: Int): ZIO[Scope, Nothing, EventLoopGroup] =
+    make(ZIO.succeed(new KQueueEventLoopGroup(nThreads)))
+
+  def epoll(nThreads: Int, executor: Executor): ZIO[Scope, Nothing, EventLoopGroup] =
+    make(ZIO.succeed(new EpollEventLoopGroup(nThreads, executor)))
+
+  def uring(nThread: Int): ZIO[Scope, Nothing, EventLoopGroup] =
+    make(ZIO.succeed(new IOUringEventLoopGroup(nThread)))
+
+  def uring(nThread: Int, executor: Executor): ZIO[Scope, Nothing, EventLoopGroup] =
+    make(ZIO.succeed(new IOUringEventLoopGroup(nThread, executor)))
+
+  def auto(nThreads: Int): ZIO[Scope, Nothing, EventLoopGroup] =
+    if (Epoll.isAvailable)
```

Comment:
I have seen this same logic in another place as well. I think it would be better to have this decision logic in the configuration layer (if it is set to `auto`) and in all these places just use a `ChannelTypeWithoutAuto` (with a better name) type. 

Maybe this is out of the scope of this PR though

Reply:
I removed the `auto` members from `EventLoopGroups` and `ChannelFactories` and moved the logic to `fromConfig`.

However, not sure what was being asked for in the second part of the comment:
> and in all these places just use a ChannelTypeWithoutAuto (with a better name) type.

Reply:
Yeah this is not exactly what I had in mind. What I was thinking to reduce that decision completely from the two places and move it to the `Config` reader part. So `Config` would always have a concrete resolved channel type (which cannot be `Auto`) and the two "users" can just take that decision and don't have to replicate that fallback logic themselves.